src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/InterventionService.kt

```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Intervention
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.InterventionRepository
+import java.util.UUID
+
+@Service
+class InterventionService(val repository: InterventionRepository) {
+
+  fun getIntervention(id: UUID): Intervention? {
+    return repository.findByIdOrNull(id)
+  }
+  fun getInterventionsForServiceProvider(id: String): List<Intervention> {
+    return repository.findByDynamicFrameworkContractServiceProviderId(id)
+  }
+  fun createIntervention(intervention: Intervention): Intervention {
+    return repository.save(intervention)
```

Comment:
there are quite a few entities that need to exist for a full intervention to be stored:

- ServiceCategory
- ServiceProvider
- ContractEligibility
- NPSRegion / PCCRegion
- DynamicFrameworkContract

i think we should be ensuring that all these things are persisted here. which follows the example of `ReferralService.sendDraftReferral` and `ReferralService.createDraftReferral`.

Reply:
Totally agree - as part of the CSV work (separate Pull Request) they will exist in there. It has a separate DTO too - as it requires more information than the one the UI requires. ( I am happy to remove this method for now unless you're ok with the fact that the next PR will pick this up.)